src/autumnaton.ts

```diff
@@ -0,0 +1,217 @@
+        const adjustedDropAmount =
+          (preAcuityExpectation + postAcuityExpectation) *
+          Math.min(1, maximumDrops / overallExpectedDropQuantity);
+        return adjustedDropAmount * value;
+      }
+    );
+    return seasonalItemValue(location) + expectedCollectionValue;
+  }
+}
+
+function seasonalItemValue(location: Location, seasonalOverride?: number): number {
+  // Find the value of the drops based on zone difficulty/type
+  const autumnItems = $items`autumn leaf, AutumnFest ale, autumn breeze, autumn dollar, autumn years wisdom`;
+  const avgValueOfRandomAutumnItem = garboAverageValue(...autumnItems);
+  const autumnMeltables = $items`autumn debris shield, autumn leaf pendant, autumn sweater-weather sweater`;
+  const autumnItem = getAutumnatonUniques(location)[1];
+  const seasonalItemDrops = seasonalOverride ?? AutumnAton.seasonalItems();
+  return (
+    (seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0) +
+    (autumnMeltables.includes(autumnItem)
+      ? // If we already have the meltable, then we get a random item, else value at 0
+        availableAmount(autumnItem) > 0
+        ? avgValueOfRandomAutumnItem
+        : 0
+      : garboValue(autumnItem, true))
+  );
+}
+
+function expectedRemainingExpeditions(legOverride?: number): number {
+  const availableAutumnatonTurns = estimatedTurns() - AutumnAton.turnsLeft();
+  let expeditionTurnSum = 0;
+  let quests = get("_autumnatonQuests", 0);
+  while (expeditionTurnSum < availableAutumnatonTurns) {
+    expeditionTurnSum +=
+      11 *
+      Math.max(
+        1,
+        quests -
+          (legOverride ?? AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length)
+      );
+    quests++;
+  }
+
+  return quests - get("_autumnatonQuests", 0);
+}
+export function prioritizeUpgradeLocations(fullLocations: Location[]): Location[] {
+  // This function shouldn't be getting called if we don't have an expedition left
+  if (expectedRemainingExpeditions() < 1) {
+    return fullLocations;
+  }
+  const profitRelevantUpgrades = [
+    "leftarm1",
+    "leftleg1",
+    "rightarm1",
+    "rightleg1",
+    "cowcatcher",
+    "periscope",
+    "radardish",
+  ] as const;
+  // Libram doesn't have amount of leg upgrades specifically like the other types do
+  const legUpgrades = AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length;
+  const currentUpgrades = AutumnAton.currentUpgrades();
+  const acquirableUpgrades = profitRelevantUpgrades.filter(
+    (upgrade) => !currentUpgrades.includes(upgrade)
+  );
+  const currentBestLocation = maxBy(fullLocations, (loc: Location) => averageAutumnatonValue(loc));
+  const currentExpectedProfit =
+    averageAutumnatonValue(currentBestLocation) * expectedRemainingExpeditions();
+
+  const upgradeValuations = acquirableUpgrades.map((upgrade) => {
+    const upgradeLocations = fullLocations.filter(
+      (location) => getAutumnatonUniques(location)[0] === upgrade
+    );
```

Comment:
I'm adding the autumnAton uniques to libram and it'll use an object instead of a tuple--we can burn that bridge when we come to it